src/web-push-lib.js

```diff
@@ -155,7 +155,7 @@ WebPushLib.prototype.generateRequestDetails = function(subscription, payload, op
         currentVapidDetails = options.vapidDetails;
       }
 
-      if (options.TTL) {
+      if (Number.isInteger(options.TTL) && options.TTL > -1) {
```

Comment:
```suggestion
      if (options.TTL !== undefined) {
```

And then:
```
timeToLive = Number(options.TTL)
if (timeToLive < 0) {
  throw new Error("TTL should be a number and should be at least 0");
}
```

Reply:
Sure, will add the suggested changes.

Before going ahead, I want to point out the implications this change will have on the Impression stats for people using the library.

So, currently whoever is sending TTL=0, the messages are actually getting delivered for a timespan for 4 weeks as that is the default.

**With this change, the stats being recorded by such end-users for their bulk campaigns sent using this library will drastically drop.**

Would want to hear your thoughts on the same.

Reply:
I think it's an acceptable drawback, we need to honor the choice of the users.

Reply:
Agree, this aligns the library with the VAPID [spec](https://tools.ietf.org/html/draft-ietf-webpush-protocol-12#section-5.2)